cmd/lifecycle/analyzer.go

```diff
@@ -66,6 +66,7 @@ func (a *analyzeCmd) Args(nargs int, args []string) error {
 		return cmd.FailErrCode(err, cmd.CodeForInvalidArgs, "parse arguments")
 	}
 	a.LifecycleInputs.OutputImageRef = args[0]
+	a.LifecycleInputs.AccessChecker = &platform.AccessCheckerImpl{}
```

Comment:
If we do https://github.com/buildpacks/lifecycle/pull/1024/files?show-viewed-files=true&file-filters%5B%5D=#r1124696772 we could pass the image handler in when we call `ResolveInputs()` rather than having the access checker live on this struct. This would require updating `LifecycleInputsOperation` to include an image handler, but that seems fine.

Reply:
The main reason for putting it to the struct in the first place was that it eases the testing because we can inject a mock. Our main concern was to adapt every `LifecycleInputsOperation` although only 2 are needing it.